oneanddone/users/views.py

```diff
@@ -151,6 +151,16 @@ def login_failure(self, *args, **kwargs):
             logging in, let us know by emailing <a href="mailto:{email}">{email}</a>.
         """).format(email='oneanddone@mozilla.com'), extra_tags='safe')
         return super(Verify, self).login_failure(*args, **kwargs)
+    
+    def login_success(self, *args, **kwargs):
+        if (self.request.user.is_authenticated() and
+                not UserProfile.objects.filter(user=self.request.user).exists()):
+            return redirect('users.profile.create')
+        elif (self.request.user.is_authenticated() and
+                not UserProfile.objects.get(user=self.request.user).privacy_policy_accepted):
+            return redirect('users.profile.update')
+        else:
+            return super(Verify, self).login_success(*args, **kwargs)
```

Comment:
If you only call `super(Verify)` here, then in the first two cases, where you want to redirect the user, they will not end up being authenticated at all, because that happens in the parent `Verify`. Also, as you will see if you remove the `if` statements I mentioned above, not calling the parent `Verify` will result in an error.

You need to call the parent `Verify` in all cases, and that method is what will do the redirect for you. Take a look at that to see what you need to do in this method before calling `super(Verify)` in order to redirect to the page in question.